gen/typedef.go

```diff
@@ -114,20 +114,18 @@ func typedef(g Generator, spec *compile.TypedefSpec) error {
 		</* We want the behavior of the underlying type for typedefs: in the case that
 				they are objects or arrays, we need to cast to the underlying object or array;
 				else, zapMarshaler in zap.go will take care of it. */>
-		<if (eq (zapEncoder .Target) "Object") ->
+		<if (zapTypedefHasGeneratedMarshaler .Target) ->
```

Comment:
optional: this change seems not necessary tho. We can keep as `<if (eq (zapEncoder .Target) "Object") ->` and same for the `else if`. But logically they are the same.

Reply:
You're right that this one isn't strictly necessary, but I wanted to avoid having loosly coupled if checks for `==Object || == "Array"` in `zapMarshaler` and this template over here, so it seemed like a good idea to introduce `zapTypedefHasGeneratedMarshaler`

Reply:
cool. I'm fine either way. 🚢